presto-main/src/main/java/com/facebook/presto/operator/SpatialIndexBuilderOperator.java

```diff
@@ -33,6 +32,12 @@
 public class SpatialIndexBuilderOperator
         implements Operator
 {
+    @FunctionalInterface
+    public interface SpatialPredicate
+    {
+        boolean apply(OGCGeometry probe, OGCGeometry build, Optional<Double> radius);
```

Comment:
This predicate may be a hotspot. Or may not. 
Anyhow i would try to avoid extra unboxing. You can use `OptionalDouble`, or even plain `double` and pass `-1` when radius is missing (since radius cannot be negative).